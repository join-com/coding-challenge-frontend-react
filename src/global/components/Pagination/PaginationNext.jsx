import React from 'react';
import PropTypes from 'prop-types';

import I18n from '../../../utils/I18n';

import Button from '../Button';

import './css/PaginationNext.css';

const PaginationNext = ({
  totalPages, pageNumber, onPageNumberChange,
}) => {
  if (totalPages > 1 && pageNumber < totalPages) {
    return (
      <Button
        className="PaginationNext__button"
        text={I18n.t('en.pagination.next')}
        params={{ pageNumber: pageNumber + 1 }}
        onClick={onPageNumberChange}
      />
    );
  }
  return null;
};

export default PaginationNext;

PaginationNext.propTypes = {
  totalPages: PropTypes.number,
  pageNumber: PropTypes.number,
  onPageNumberChange: PropTypes.func,
};

PaginationNext.defaultProps = {
  totalPages: 0,
  pageNumber: 1,
  onPageNumberChange: () => {},
};
