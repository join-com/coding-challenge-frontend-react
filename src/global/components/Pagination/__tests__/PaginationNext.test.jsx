import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PaginationNext from '../PaginationNext';

const props = {
  totalPages: 3,
  pageNumber: 2,
};

describe('PaginationNext', () => {
  it('should render the component for totalPages > 1 & pageNumber < totalPages', () => {
    const tree = shallow(<PaginationNext {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should not render the component for pageNumber = totalPages', () => {
    const tree = shallow(<PaginationNext {...props} pageNumber={3} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should not render the component for totalPages = 1', () => {
    const tree = shallow(<PaginationNext {...props} totalPages={1} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
