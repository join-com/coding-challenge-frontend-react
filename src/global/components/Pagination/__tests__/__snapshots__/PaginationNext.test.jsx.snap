// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationNext should not render the component for pageNumber = totalPages 1`] = `""`;

exports[`PaginationNext should not render the component for totalPages = 1 1`] = `""`;

exports[`PaginationNext should render the component for totalPages > 1 & pageNumber < totalPages 1`] = `
<ButtonWrapper
  className="PaginationNext__button"
  onClick={[Function]}
  params={
    Object {
      "pageNumber": 3,
    }
  }
  text="Next"
/>
`;
